frontend/packages/console-shared/src/components/actions/LazyActionMenu.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import { MenuToggle } from '@patternfly/react-core';
+import { EllipsisVIcon } from '@patternfly/react-icons';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { Action } from '@console/dynamic-plugin-sdk';
+import { checkAccess } from '@console/internal/components/utils';
+import ActionServiceProvider from './ActionServiceProvider';
+import ActionMenuRenderer from './menu/ActionMenuRenderer';
+import { ActionContext, ActionMenuVariant } from './types';
+
+type LazyActionMenuProps = {
+  context: ActionContext;
+  variant?: ActionMenuVariant;
+  label?: string;
+  isDisabled?: boolean;
+};
+
+const LazyActionMenu: React.FC<LazyActionMenuProps> = ({
+  context,
+  variant = ActionMenuVariant.KEBAB,
+  label,
+  isDisabled,
+}) => {
+  const { t } = useTranslation();
+  const isKebabVariant = variant === ActionMenuVariant.KEBAB;
+  const [active, setActive] = React.useState<boolean>(false);
+  const [initActionLoader, setInitActionLoader] = React.useState<boolean>(false);
+  const toggleRef = React.useRef<HTMLButtonElement>();
+  const toggleRefCb = React.useCallback(() => toggleRef.current, []);
+  const toggleLabel = label || t('console-shared~Actions');
+
+  const toggleMenu = () => setActive((value) => !value);
+
+  const hideMenu = () => {
+    toggleRef.current?.focus();
+    setActive(false);
+  };
+
+  const handleRequestClose = (e?: MouseEvent) => {
+    if (!e || !toggleRef.current?.contains(e.target as Node)) {
+      hideMenu();
+    }
+  };
+
+  const handleHover = React.useCallback(() => {
+    setInitActionLoader(true);
+  }, []);
+
+  const runAccessChecks = (actions: Action[]) => {
+    // Check access after loading actions from service over a kebab to minimize flicker when opened.
+    // This depends on `checkAccess` being memoized.
+    _.each(actions, (action: Action) => {
+      if (action.accessReview) {
+        checkAccess(action.accessReview);
```

Comment:
This seems like an odd practice. Wouldn't this run the risk of throwing an error and crashing the page? We do it in other places which is why I said it was an odd practice.

Also, without doing anything with the response, I'm a bit confused on how this is doing anything 🤔

Reply:
Upon further thought --- I guess this is async, so it wouldn't blow up the main thread if it fails... still unsure how it helps though.

Reply:
Yeah, the action access checks are async. We do this upfront to minimise the flicker while running these access checks when menu is actually opened. So when users hover on the kebab we run these checks and memoize them. That way when action menu is actually opened it uses those memoized results rather than making another call.